lock-api-objects/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -114,19 +116,21 @@ default List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTr
     long currentTimeMillis();
 
     /**
-     * Acquires a lease on a named timestamp. The lease is taken out with a new fresh timestamp.
-     * The returned timestamps are fresh timestamps obtained strictly after the lease is taken out.
+     * Acquires a lease on named timestamps. The lease is taken out with a new fresh timestamp.
+     * The timestamps supplied are fresh timestamps obtained strictly after the lease is taken out.
+     * The supplier returns exactly the number of timestamps requested and throws on any additional
+     * interactions.
```

Comment:
A general philosophy question: I think this specification should exist on the underlying `TimestampLeaseResult` object? Not clear where there's a supplier here (or, if we change the implementation of `TimestampLeaseResults` for some reason, then there may not be an actual supplier.)